### ISSUE TITLE
Markdown after HTML-block without blank line (pegdown-migration)

### ISSUE THREAD

**Issue by Elmervc** [vsch/flexmark-java#115]
edit: I've edited the issue to reflect the more general case.

We have many cases in user entered data where an HTML-block is directly followed by a text-line with markdown in it (e.g. a line with markdown link or header line).

Input:
```
<hr/>
### Some header
```

Commonmark output:
```
<hr/>
### Some header
```
Most other libs, including pegdown:
```
<hr/>

<h3>Some header</h3>
```
Pegdown (and many others) allow markdown text directly after an HTML-block ([babelmark here](https://johnmacfarlane.net/babelmark2/?text=%3Chr%2F%3E%0A%23%23%23+Some+header)).

Commonmark won't parse it as markdown by spec:
>**HTML blocks**
>...
>6. 
>**Start condition:** line begins the string < or </ followed by one of the strings (case-insensitive) address, article, ... , hr, ..., track, ul, followed by whitespace, the end of the line, the string >, or the string />.
>**End condition:** line is followed by a blank line.

Is there some way to enable markdown parsing of a line directly after a closing condition of a HTML-block? When the option is there, it should be enabled in the Pegdown options-adapter.

**Comment by vsch** [comment]
@Elmervc, the `<hr/>` opens an HTML block that extends to the next blank line. So either a blank line must be added between the HTML block and the header or `----` or `***` thematic rule should be used instead of the HTML tag to get the horizontal rule.

The blocks first inlines second parser architecture has limitations which it trades off for parsing speed.

Why is an HTML tag used instead of the markdown element when it is available?

**Comment by vsch** [comment]
@Elmervc, when I implement HTML block parsing option this issue could be addressed but it will have to have a separate option to allow HTML block closing when no open tags remain, without needing a blank line after it.

**Comment by Elmervc** [comment]
@vsch 

> Why is an HTML tag used instead of the markdown element when it is available?
It's user input we don't control ;) I had to fix 70+ instances of this issue (markdown after HTML without blank line) in our app-data, which I scanned with some regexes. Most cases were unparsed markdown links in lines not preceded by a blank line after HTML.

**Comment by vsch** [comment]
@Elmervc, typical case of users will use whatever works by trial and error. Emulation usually requires implementing all idiosyncrasies and bugs to be bullet proof. Flexmark will never be a 100% compatible replacement for pegdown due to parser architecture differences.

The HTML parsing will improve the compatibility but will not solve all issues.

**Comment by vsch** [comment]
@Elmervc, this too is fixed in 0.22.2 in pegdown compatibility mode.
